CIP-0095/README.md

```diff
@@ -0,0 +1,678 @@
+containing a DRep retirement certificate. The wallet should articulate this
+request from client application in a explicit and highly informative way. Users
+should be made aware of the type of certificate, associated DRepID and metadata
+anchor.
+
+If user grants permission, the transaction must be signed by the secret key of
+the provided public DRep Key, with the signature and key to be added to the
+transaction witness set before submission.
+
+By forcing wallets to inspect these certificates it allows the user to catch
+malicious client applications attempting to insert their own data into the
+certificate.
+
+##### Errors
+
+One `TxSignError` should be returned if there is a signature error with any of
+the certificates.
+
+##### Returns
+
+This returns a `SubmittedTransaction` object which contains all the details of
+the submitted retirement certificate, for the client to confirm. The returned
+`txHash` can be used by the client to track the status of the transaction
+containing the certificate on Cardano.
+
+#### `api.submitVote([tx: UnsignedTransaction, dRepKey: PubDRepKey]): Promise<SubmittedTransaction>[]`
+
+Errors: [APIError](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+This endpoint requests the wallet to inspect, sign and submit transaction(s)
+containing votes. The wallet should articulate this request from client
+application in a explicit and highly informative way. For each vote, users must
+be shown the governance action ID, vote choice (yes, no, abstain) and metadata
+anchor. For the case of multiple votes at once, the user may only be asked for
+permission to authorize all at once, but this can depend on wallet's
+implementation.
+
+If user grants permission, each transaction must be signed by the secret key of
+the provided public stake key, with the signature and key to be added to the
+transaction witness set before submission.
+
+By forcing wallets to inspect these votes it allows the user to catch malicious
+client applications attempting to alter the vote's contents.
+
+##### Errors
+
+One `TxSignError` should be returned if there is a signature error with any of
+the transactions.
+
+##### Returns
+
+This returns an array of `SubmittedTransaction` objects which contain the
+details of the submitted votes, for the client to confirm. The returned
+`txHash`s can be used by the client to track the status of the transactions
+containing the certificates on Cardano.
+
+#### `api.submitGovernanceAction(tx: UnsignedTransaction): Promise<SubmittedTransaction>`
+
+Errors: [APIError](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+This endpoint requests the wallet to inspect, sign and submit a transaction
+containing a governance action. The wallet should articulate this request from
+client application in a explicit and highly informative way. For the governance
+action the user must be shown the amount of ADA to be locked as deposit,
+governance action type and the metadata anchor.
```

Comment:
This is comprehensive for the Info and Motion of No-Confidence governance action types, however all of the other governance action types require some form of input from the  user submitting the governance action (the Proposal Submitter). For example in the case of a Treasury Withdrawal a Proposal Submitter would need to provide an amount (in Lovelace) and an address where that amount would be sent if the governance action were to be enacted. 

I think it would improve the security of the process if these input names and values were also displayed to the Proposal Submitter at this point by the wallet so that the Proposal Submitter could verify that an app creating such a transaction was submitting the correct/expected information to the wallet.

Reply:
Good suggestion, would add burden to wallet implementers but I think it is likely necessary.

Reply:
Added in [9af3bf0](https://github.com/cardano-foundation/CIPs/pull/509/commits/9af3bf045daf57bcb46116b69f446db685fb9a79).